AsyncDisplayKit/Details/_ASDisplayView.mm

```diff
@@ -375,3 +376,81 @@ - (UIView *)preferredFocusedView
 }
 #endif
 @end
+
+
+#pragma mark - Accessibility
+
+static const char *ASDisplayNodeAssociatedNodeKey = "ASAssociatedNode";
+
+@implementation UIAccessibilityElement (_ASDisplayView)
+
+- (void)setAsyncdisplaykit_node:(ASDisplayNode *)node
+{
+  objc_setAssociatedObject(self, ASDisplayNodeAssociatedNodeKey, node, OBJC_ASSOCIATION_ASSIGN); // Weak reference to avoid cycle, since the node retains the layer.
+  
+  // Update UIAccessibilityElement properties from node
+  self.accessibilityLabel = node.accessibilityLabel;
+  self.accessibilityHint = node.accessibilityHint;
+  self.accessibilityValue = node.accessibilityValue;
+  self.accessibilityTraits = node.accessibilityTraits;
+}
+
+- (ASDisplayNode *)asyncdisplaykit_node
+{
+  return objc_getAssociatedObject(self, ASDisplayNodeAssociatedNodeKey);
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  ASDisplayNode *asyncDisplayKitNode = self.asyncdisplaykit_node;
+  if ( _accessibleElements != nil ) {
+    return _accessibleElements;
+  }
+  
+  _accessibleElements = [[NSMutableArray alloc] init];
+  
+  // Create UI accessiblity elements for each subnode that represent the subnode within the accessibility container
+  for (ASDisplayNode *subnode in asyncDisplayKitNode.subnodes) {
+    if (subnode.isAccessibilityElement || [subnode accessibilityElementCount] > 0) {
+      UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:self];
+      accessibilityElement.asyncdisplaykit_node = subnode;
+      [_accessibleElements addObject:accessibilityElement];
+    }
+  }
+  
+  return _accessibleElements;
+}
+
+- (NSInteger)accessibilityElementCount
+{
+  return [self accessibleElements].count;
+}
+
+- (id)accessibilityElementAtIndex:(NSInteger)index
+{
+  UIAccessibilityElement *element = [self accessibleElements][index];
```

Comment:
If we do cache this, I wonder if there is a way to keep the array for only the amount of time it takes to complete a certain batch of calls to these APIs?

Even with this call point though, because the correct way to invalidate the cash would be to use the environment, it probably makes sense to have no optimization for now and then go directly to the environment based implementation if we can measure the overhead in a way that affects the user experience